vfs/read.go

```diff
@@ -62,7 +65,11 @@ func (fh *ReadFileHandle) openPending() (err error) {
 	if fh.opened {
 		return nil
 	}
-	r, err := fh.o.Open()
+	var end int64 = -1
+	if fh.chunkSize > 0 {
+		end = fh.chunkSize - 1
```

Comment:
This should probably cap end to o.Size(), or maybe add another condition in the if `fh.chunkSize > 0 && fh.chunkSize < o.Size()`?